jobs/delorean/jenkinsfiles/2.0/ews/discovery/umb/Jenkinsfile

```diff
@@ -52,8 +68,12 @@ pipeline {
 
         stage('Push Branch') {
             steps {
-                //ToDo Push the product branch if an changes have been made
-                echo "Push Branch"
+                script {
+                    def changes = sh(script: 'git log origin/${PRODUCT_BRANCH}..HEAD --pretty=o | wc -l', returnStdout: true)
+                    if ((changes as int) > 0) {
+                        sh 'git push origin ${PRODUCT_BRANCH} --force'
```

Comment:
Would there need to be a commit in here before the push?

Reply:
I'm not sure really, should we commit here or in the process step where we know what changes need to be made? maybe revisit when it's more fleshed out?

Reply:
Each stage will deal with committing changes that it makes. This stage should only push the branch if there are changes to push.